lib/human.py

```diff
@@ -0,0 +1,146 @@
+import io
+import typing
+
+from . import logic, manual_testing
+
+
+# TODO: how should a transpiler handle unexpected values? logging.warning? Throw an exception? Leave a TODO comment in the output?
+
+def main():
+    # path = 'inputs/inputFile.edn'
+    # path = 'inputs/jamestest.edn'
+    path = 'inputs/tqqq_long_term.edn'
+    root_node = manual_testing.get_root_node_from_path(path)
+    print(convert_to_pretty_format(root_node))
+
+
+def pretty_fn(fn_string: str) -> str:
+    if fn_string == logic.ComposerIndicatorFunction.RSI:
+        return "RSI"
+    if fn_string == logic.ComposerIndicatorFunction.CURRENT_PRICE:
+        return "Close"
+    if fn_string == logic.ComposerIndicatorFunction.CUMULATIVE_RETURN:
+        return "CumulativeReturn"
+    if fn_string == logic.ComposerIndicatorFunction.MOVING_AVERAGE_PRICE:
+        return "SMA"
+    print(f"UNEXPECTED function {fn_string}")
+    return fn_string
+
+
+def pretty_comparison(comparator_string: str) -> str:
+    if comparator_string == logic.ComposerComparison.LTE:
+        return "<="
+    if comparator_string == logic.ComposerComparison.LT:
+        return "<"
+    if comparator_string == logic.ComposerComparison.GTE:
+        return ">="
+    if comparator_string == logic.ComposerComparison.GT:
+        return ">"
+    if comparator_string == logic.ComposerComparison.EQ:
+        return "="
+    print(f"UNEXPECTED comparator {comparator_string}")
+    return comparator_string
+
+
+def pretty_indicator(fn: str, val, window_days: typing.Optional[int]):
+    if fn == logic.ComposerIndicatorFunction.CURRENT_PRICE:
+        return f"{pretty_fn(fn)}({val})"
+    else:
+        return f"{pretty_fn(fn)}({val}, {window_days}d)"
+
+
+def pretty_lhs(node) -> str:
+    if type(node[":lhs-val"]) != str:
+        return node[":lhs-val"]
+    else:
+        return pretty_indicator(node[":lhs-fn"], node[":lhs-val"], node.get(":lhs-window-days"))
+
+
+def pretty_rhs(node) -> str:
+    if node.get(':rhs-fixed-value?', type(node[':rhs-val']) != str):
+        return node[':rhs-val']
+    else:
+        return pretty_indicator(node[":rhs-fn"], node[":rhs-val"], node.get(":rhs-window-days"))
+
+
+def pretty_condition(node) -> str:
+    return f"{pretty_lhs(node)} {pretty_comparison(node[':comparator'])} {pretty_rhs(node)}"
+
+
+def pretty_selector(node) -> str:
+    if node[":select-fn"] == ":bottom":
+        return f"bottom {node[':select-n']}"
+    elif node[":select-fn"] == ":top":
+        return f"top {node[':select-n']}"
+    else:
+        return f"UNEXPECTED :select-fn {node[':select-fn']}"
+
+
+def pretty_filter(node) -> str:
+    return f"{pretty_selector(node)} by {pretty_indicator(node[':sort-by-fn'], '____', node[':sort-by-window-days'])}"
+
+
+def print_children(node, depth=0, parent_node_branch_state: typing.Optional[logic.NodeBranchState] = None, file=None):
```

Comment:
This indents for every level, but in VectorBT, when we get passed the data fetching and indicator precomputing, we only want to indent after some nodes (:if-child nodes).

My point is this logic here is not totally reusable. Hence my current aversion to OOP and overriding methods. If I find myself repeating something *and* it's something I shouldn't be repeating, then I'll refactor.